### DISCUSSION TITLE
Outflow boundary condition

### DISCUSSION THREAD

**Discussion by Alex-jian522** [AMReX-Combustion/PeleLMeX#285] [General]
Hello, everyone!

I am currently running a 2D simulation using PeleLMEx for a jet flow towards a wall. I have set the boundary conditions on the right and left sides to be outflow.

Everything looks good when I set peleLM.gravity to 0 0 0. However, when I change peleLM.gravity to 0 -9.81 0, I am noticing some unexpected flow entering the computational domain from the outflow boundary. I would like to understand the composition and properties of this flow and also determine whether this behavior is correct for the outflow boundary condition.

Thank you very much for any assistance or insights on this matter!
![streamline_06000](https://github.com/AMReX-Combustion/PeleLMeX/assets/72745362/b2cba10e-08c6-4946-943e-f15408ec25ea)

**Comment by Alex-jian522** [comment]
Upon examining the composition, I guess that that strange inflow consists of air, which matches the initial condition specified for the computational domain. However, I have observed that the temperature of the inflow is not 300K, as set in the initial condition.

**Comment by drummerdoc** [comment]
The problem I believe is that the hydrostatic BC is not properly set up in LMeX yet.  It's on a list somewhere of things to do very soon.  If you watch an animation, you'll probably see your flow dumping out the side Bus

**Reply by Alex-jian522** [reply]
Thank you very much for your prompt reply!

I would like to inquire about the projected timeline for resolving this issue. Could you kindly provide an estimate of whether it might take weeks or possibly months to find a solution? Additionally, I am considering whether I can switch to using PeleLM instead of PeleLMeX. Would PeleLM be a viable alternative and potentially offer a solution to the problem at hand? I would greatly appreciate any insights or guidance that you can provide.

**Comment by drummerdoc** [comment]
In your case, LM might be a good short term option.  When LMeX is ready, the transition would be easy.  The three main reasons to avoid LM are 1. Subcycling is MUCH more complicated, 2. LM is no longer being actively updated, and 3. If you had a closed system, LM can't manage the pressure rise correctly.

However, the inputs and problem setup are nearly identical so the transition would be simple.

As for schedule for the BCs in LMeX, I honestly don't know.  There are are several things high on our list to complete before the project that funds both codes (and PeleC) ends (12/31/2023) and the project is quickly ramping down and losing staff.  For LMeX, the higher priorities are robust solvers and more complex geometries, along with these BC problems.  For PeleC, it's NSCBC outflow.  And for both it's portability to the remaining platforms.

**Comment by hong27** [comment]
Hello, @drummerdoc Could you please advise if this hydrostatic outflow BC issue has been addressed? If not yet, can we expect it to be fixed in the near future? We are planning a simulation of buoyant flows. If such a fix is not easily achieved soon, I think we will need to consider some workarounds, such as using wall boundary conditions instead of the outflow BC. Thanks!

**Comment by nickwimer** [comment]
@hong27 The hydrostatic outflow is not fully implemented/tested in LMeX just yet. Are you looking to perform a simulation of a buoyant flow with all open boundaries or the simulation that you outlined at the top of this thread?

**Reply by hong27** [reply]
@nickwimer Thanks for the prompt reply. 

The simulation that we are planning is something like:
`peleLM.lo_bc = Outflow Interior Inflow` 
`peleLM.hi_bc = Outflow Interior Outflow`
It would be great if the hydrostatic outflow issue is fixed soon. If this is not possible, I think we might need to consider an alternative simulation with
`peleLM.lo_bc = NoSlipWallAdiab Interior Inflow` 
`peleLM.hi_bc = NoSlipWallAdiab Interior Outflow`,
with the domain in the x-direction extended. This may not be the best idea though.

Does PeleLMeX have an estimated time by which this boundary condition issue could be addressed? 

Thank you very much for your help in this.
